git_repo/services/ext/gitbucket.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python
+
+import logging
+log = logging.getLogger('git_repo.gitbucket')
+
+from ..service import register_target
+from .github import GithubService
+
+import github3
+
+@register_target('bucket', 'gitbucket')
+class GitbucketService(GithubService):
+    fqdn = "localhost"
+    port = 8080
+    def __init__(self, *args, **kwarg):
+        super(GitbucketService, self).__init__(*args, **kwarg)
+
+    def connect(self):
+        self.gh = github3.GitHubEnterprise(self.build_url(self))
```

Comment:
ok, but that's going to be an issue. The `self.gh` instance shall be created in the constructor as first line [even before the call the the constructor](https://github.com/guyzmo/git-repo/blob/devel/git_repo/services/ext/github.py#L20,L22).

The reason for that is the tests, where it pulls [the `session` instance out of the `self.gh` instance](https://github.com/guyzmo/git-repo/blob/devel/tests/integration/test_github.py#L35,L36). If you change the `self.gh` instance at `self.connect()` time, there's a high probability that the tests will be testing another session instance than the one you're actually using in your methods.

So, we could patch [GithubService.__init__()](https://github.com/guyzmo/git-repo/blob/devel/git_repo/services/ext/github.py#L21,L21) so that:

````python
GITHUB_FQDN='github.com'

@register_target('hub', 'github')
class GithubService(RepositoryService):
    fqdn = GITHUB_FQDN

    def __init__(self, *args, **kwarg):
        if self.fqdn == GITHUB_FQDN:
            self.gh = github3.GitHub()
       else:
            self.gh = github3.GitHubEnterprise(RepositoryService.build_url(self))
        super(GithubService, self).__init__(*args, **kwarg)
```

and as a side effect, there's a chance that actually it enables `GithubService` to work with self hosted github instances… out of the box 🙌